Classes/Session/Session.php

```diff
@@ -0,0 +1,136 @@
+<?php
+namespace EWW\Dpf\Session;
+
+/*
+ * This file is part of the TYPO3 CMS project.
+ *
+ * It is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License, either version 2
+ * of the License, or any later version.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ *
+ * The TYPO3 project - inspiring people to share!
+ */
+
+class Session
+{
+    const ROOT_KEY = "tx_dpf";
+    const LIST_ACTION_KEY = "list_action";
+    const WORKSPACE = "workspace";
+
+
+    /**
+     * @return SearchSessionData $data
+     */
+    public function getWorkspaceData()
+    {
+        $sessionData = $this->getData(self::ROOT_KEY);
+        if (array_key_exists(self::WORKSPACE, $sessionData)) {
+            return unserialize($sessionData[self::WORKSPACE]);
+        }
+
+        return new SearchSessionData();
+    }
+
+    /**
+     * @param SearchSessionData $data
+     */
+    public function setWorkspaceData(SearchSessionData $data)
+    {
+        $sessionData[self::WORKSPACE] = serialize($data);
+        $this->setData($sessionData);
+    }
+
+
+    /**
+     * Stores the given action name, controller name and uri.
+     *
+     * @param $action
+     * @param $controller
+     */
+    public function setListAction($action, $controller, $uri = null)
+    {
+        $sessionData = $this->getData(self::ROOT_KEY);
+        $sessionData[self::LIST_ACTION_KEY] = [$action, $controller, $uri];
+        $this->setData($sessionData);
+    }
+
+    /**
+     * Returns the stored action name, controller name and uri.
+     * [ 0 => 'action name', 1 => 'controller name', 2 => 'uri']
+     *
+     * @return array|mixed
+     */
+    public function getListAction()
```

Comment:
[naming]
So that's "getStoredActionURI" ?

Reply:
It is not only the URI the action and controller name are also stored. I renamed it into getStoredAction.